internal/ackhandler/sent_packet_handler.go

```diff
@@ -455,15 +458,19 @@ func (h *sentPacketHandler) SendMode() SendMode {
 		h.logger.Debugf("Limited by the number of tracked packets: tracking %d packets, maximum %d", numTrackedPackets, protocol.MaxTrackedSentPackets)
 		return SendNone
 	}
-	// Send retransmissions first, if there are any.
-	if len(h.retransmissionQueue) > 0 {
-		return SendRetransmission
+	if h.numRTOs > 0 {
+		h.numRTOs--
```

Comment:
This makes this function non-pure, which isn't clear from the name. Either change the name or do the state change where you actually send the RTO.

Reply:
@lucas-clemente `NextPacketSendMode`?

Reply:
That doesn't solve the issue. I think the best fix is decrementing numRTOs when actually queueing them.

Sad that Go doesn't support const functions 🙄